### ISSUE TITLE
Panic, not error, when opening a locked or improperly closed DB

### ISSUE THREAD

**Issue by 0joshuaolson1** [akrylysov/pogreb#12]
I ran this (on Linux) twice:
```
package main
import(
 DB "github.com/akrylysov/pogreb"

 "fmt"
)
func main(){
 db, err := DB.Open("test.db", &DB.Options{0, nil})
 db=db
 fmt.Println(err)
 //db.Close()
}
```
The first run creates a lock file. The second panics with:
```
pogreb: Performing recovery...
pogreb: Index file size=1024; data file size=512
pogreb: Header dbInfo {level:0 count:0 nBuckets:1 splitBucketIdx:0 freelistOff:-1 hashSeed:2342761507}
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/akrylysov/pogreb/fs.(*osfile).Slice(0xc4200ee320, 0x20000000000, 0x20000000200, 0x200, 0x200, 0xc420055cd0)
	/home/xenox/go/src/github.com/akrylysov/pogreb/fs/os.go:60 +0x60
github.com/akrylysov/pogreb.(*bucketHandle).read(0xc4200559c8, 0x0, 0x0)
	/home/xenox/go/src/github.com/akrylysov/pogreb/bucket.go:75 +0x56
github.com/akrylysov/pogreb.(*DB).forEachBucket(0xc420108000, 0xfff00000ffffffff, 0xc420055ce0, 0xc420055ce8, 0x0)
	/home/xenox/go/src/github.com/akrylysov/pogreb/db.go:149 +0xc4
github.com/akrylysov/pogreb.recoverSplitCrash(0xc420108000, 0x0, 0x0)
	/home/xenox/go/src/github.com/akrylysov/pogreb/recovery.go:60 +0xbf
github.com/akrylysov/pogreb.(*DB).recover(0xc420108000, 0x400, 0x0)
	/home/xenox/go/src/github.com/akrylysov/pogreb/recovery.go:140 +0x2e1
github.com/akrylysov/pogreb.Open(0x694821, 0x7, 0xc420055f60, 0x0, 0xc42000e020, 0x694b29)
	/home/xenox/go/src/github.com/akrylysov/pogreb/db.go:106 +0x66b
main.main()
	/home/xenox/y.go:8 +0x5e
exit status 2
```
Everything is fine (no error either) if either
- the first run includes the `db.Close()`, or
- I manually remove the lock file between the first and second runs

**Comment by 0joshuaolson1** [comment]
With a different program it performed recovery successfully ("`Recovery Complete`"), so some unwanted output isn't necessarily from panicking.

**Comment by akrylysov** [comment]
Fixed in 0c598cc122414010d1ca5cfd06f27f735646bf3c.
